### PR TITLE
MARKET-163 added old code

### PR DESCRIPTION
**Pull request by ivy-lmu**

### CODE REVIEW

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/SampleProcessTest.java [hunk 1]
```diff
@@ -7,11 +7,10 @@
 
 @IvyProcessTest
 public class SampleProcessTest{
```
Comment (alexsuter) [inline ldap-connector-test/src_test/com/axonivy/connector/ldap/test/SampleProcessTest.java:9]: you can delete the sampleprocess

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  @Test
+  void admin_query() throws Exception {
+    NamingEnumeration<SearchResult> resultEnum = ldapQuery.perform("DC=zugtstdomain,DC=wan", "(name=admin)",
+            SearchControls.ONELEVEL_SCOPE,
+            returningAttributes);
+    assertThat(resultEnum.hasMoreElements()).isTrue();
+  }
+
+  @Test
+  void query2() throws Exception {
```
Comment (alexsuter) [inline ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java:71]: query2 is not a meaningful name

File: ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    NamingEnumeration<SearchResult> resultEnum = ldapQuery.perform("DC=zugtstdomain,DC=wan",
+            "(objectClass=*)",
+            SearchControls.SUBTREE_SCOPE,
+            returningAttributes);
+
+    while (resultEnum.hasMoreElements()) {
+      SearchResult r = resultEnum.next();
+      System.out.println(r);
+      System.out.println(r.getAttributes().get("name"));
+    }
```
Comment (alexsuter) [inline ldap-connector-test/src_test/com/axonivy/connector/ldap/test/TestLdap.java:81]: this test makes no assertions (assertThat()).
system.out.println() not needed

File: ldap-connector/config/variables.yaml [hunk 4]
```diff
@@ -6,4 +6,7 @@
 # '<project>/Config/_<environment>/variables.yaml
 #
 Variables:
-#  myVariable: value
+  LdapConnector:
+    Username: ""
+    Password: ""
+    Url: "ldap://test-ad.ivyteam.io:389"
```
Comment (alexsuter) [inline ldap-connector/config/variables.yaml:12]: this is the variables yaml of your real product.
put test data in variables.yaml of your test project.
the variables will be overridden.

the URL here must be empty
and you can URL in your test project has the test url.

File: ldap-connector/config/variables.yaml [hunk 5]
```diff
@@ -6,4 +6,7 @@
 # '<project>/Config/_<environment>/variables.yaml
```
Comment (alexsuter) [inline ldap-connector/config/variables.yaml:6]: we need to add more meta infos by adding description to all of these variables. but we can make that at the end.

File: ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+import javax.naming.InvalidNameException;
+import javax.naming.NameClassPair;
+import javax.naming.directory.SearchResult;
+import javax.naming.ldap.LdapName;
+
+import org.apache.commons.lang3.StringUtils;
+
+import ch.ivyteam.naming.JndiProvider;
+
+/**
```
Comment (alexsuter) [inline ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java:15]: remove javadoc.

File: ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  }
+
+  public String getPassword() {
+    return password;
+  }
+
+  public void setDefaultContext(String defaultContext) {
+    this.defaultContext = defaultContext;
+  }
+
```
Comment (alexsuter) [inline ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java:94]: remove javadoc

File: ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+    private JndiProvider provider = new JndiProvider(null, null);
+    private String url = "";
+    private String authenticationKind = "none";
+    private String userName = "";
+    private String password = "";
+    private String defaultContext = "";
+    private boolean useSsl = false;
+    private boolean useLdapConnectionPool = false;
+
+    public Builder provider(@SuppressWarnings("hiding") JndiProvider provider) {
```
Comment (alexsuter) [inline ldap-connector/src/com/axonivy/connector/ldap/util/JndiConfig.java:236]: I usually add @SuppressWarnings("hiding") on the builder class itself.
so you don't need to add to every method.

File: ldap-connector/src/com/axonivy/connector/ldap/LdapQueryBeanRS.java [hunk 9]
```diff
@@ -0,0 +1,665 @@
...
+ * PI-Element to query LDAP-Servers A variant from LdapQueryBean that returns
+ * the values as Recordset
+ *
+ * @author Bruno Buetler
+ * @version MarcWillaredt August2009 Updated to Ivy 4.1
+ * @version bb 16.11.2005 improved (allow "in.x" as ivy attribute names) and
+ *          extended (anyFilter)
+ * @version bb 2.11.2005 created.
+ */
+public class LdapQueryBeanRS extends AbstractUserProcessExtension {
```
Comment (alexsuter) [inline ldap-connector/src/com/axonivy/connector/ldap/LdapQueryBeanRS.java:52]: remove this extends AbtractUserProcessExtension

**Review by alexsuter — CHANGES_REQUESTED** [review]
- I would remove all comments/javadoc.
- Remove all commented code
- I think we need to refactor the LdapQueryBeanRS.

File: ldap-connector-test/pom.xml [hunk 10]
```diff
@@ -30,6 +30,10 @@
         <configuration>
           <ivyVersion>[9.3.0,]</ivyVersion>
           <engineDownloadUrl>https://developer.axonivy.com/permalink/nightly/axonivy-engine.zip</engineDownloadUrl>
+          <additionalVmOptions>
+            -Divy.Applications.ldapconnectortest.Variables.LdapConnector.Username=${activedirectory.username}
+            -Divy.Applications.ldapconnectortest.Variables.LdapConnector.Password=${activedirectory.password}
```
Comment (ivy-lmu) [inline ldap-connector-test/pom.xml:35]: @alexsuter tests are still failing on github: https://github.com/axonivy-market/ldap-connector/runs/4331195098?check_suite_focus=true

I assume that we do not override the variables

Comment (alexsuter) [reply ldap-connector-test/pom.xml:35]: Could be possible. II think we have the wrong key. Maybe the application name?
ldapconnectortest. let me check the build log

Comment (alexsuter) [reply ldap-connector-test/pom.xml:35]: hmm.. maybe the application name is "test". I would give it a try.

If this is not the case. We can shortly print the name of the application with a System.out.println(IApplication.current().getName());

Then we can see in the build log on GitHub whats the name.

**Pull request merged**